### DISCUSSION TITLE
.nrepl-port file generation

### DISCUSSION THREAD

**Discussion by rafaeldelboni** [babashka/babashka#946] [Ideas]
WDYT about an nrepl argument that, if set, generate the `.nrepl-port` port in the folder the `bb nrepl` was invoked?
And maybe, if possible, delete when the repl session is over.

**Comment by borkdude** [comment]
We had some discussion about this regarding the socket REPL here: https://github.com/babashka/babashka/issues/157.
One of the issues might be that this conflicts with a JVM nREPL instance. But if we would introduce an argument for this, like you suggest, this behavior would be more explicit and probably safe to do.

**Reply by rafaeldelboni** [reply]
Usually, when the BB nrepl-port file is useful for me is when I'm running bb outside an clojure project folder and would be easier to me add an arg in the command line than manually asking to my editor connect to an nrepl.

Regarding the conflict, I wouldn't mind to introduce a new file name pattern like `.nrepl-port-bb` and open a PR on Conjure supporting this.

**Reply by borkdude** [reply]
You can also write a script for this: there is one such script in the book at the nREPL section:

https://book.babashka.org/#_nrepl

**Reply by rafaeldelboni** [reply]
Thanks, that works for me, sorry I didn't saw that.

**Reply by borkdude** [reply]
No worries :)
